base/src/main/java/org/mozilla/jss/provider/javax/crypto/JSSTrustManager.java

```diff
@@ -131,25 +182,45 @@ public void checkSignature(X509Certificate cert, X509Certificate[] caCerts) thro
         }
 
         if (issuer == null) {
-            throw new CertificateException("Unable to validate signature: " + cert.getSubjectX500Principal());
+            logger.debug("JSSTrustManager: Untrusted issuer: " + cert.getIssuerX500Principal());
+
+            // TODO: fix depth param
+            status.addReason(ValidityStatus.UNTRUSTED_ISSUER, cert, 0);
+
+            return;
         }
 
-        logger.debug("JSSTrustManager: cert signed by " + issuer.getSubjectX500Principal());
+        logger.debug("JSSTrustManager: Trusted issuer: " + issuer.getSubjectX500Principal());
     }
 
-    public void checkValidityDates(X509Certificate[] certChain) throws Exception {
+    public void checkValidityDates(X509Certificate[] certChain, ValidityStatus status) throws Exception {
 
         for (X509Certificate cert : certChain) {
 
             logger.debug("JSSTrustManager: Checking validity dates of cert 0x" + cert.getSerialNumber().toString(16));
             logger.debug("JSSTrustManager: - not before: " + cert.getNotBefore());
             logger.debug("JSSTrustManager: - not after: " + cert.getNotAfter());
 
-            cert.checkValidity();
+            try {
+                cert.checkValidity();
+
+            } catch (CertificateNotYetValidException e) {
+                logger.debug("JSSTrustManager: Cert not yet valid: " + cert.getSubjectX500Principal());
+
+                // NSS uses EXPIRED_CERTIFICATE for this case in CERT_CheckCertValidTimes()
+                // TODO: fix depth param
```

Comment:
We are following the chain from the root so depth start from `certChain.length - 1` and go to zero. It seems an easy fix, is there a reason to move this to a future PR?

Reply:
According to the doc the order of the certs in the original chain is not defined, and the chain could even be incomplete:
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/javax/net/ssl/X509TrustManager.html

In the beginning of `checkCertChain()` we sort the certs, but when calling `addReason()` we probably should use the original index (which I'm not sure what we should use if the chain is incomplete).

I think ideally `addReason()` should not need to take a `depth` param since it already has the `cert`. Maybe we can fix that later, but for now let me see if I can add another method to find the cert's index in the original (unsorted) cert chain.

Reply:
OK, I was not considering incomplete chain. Thanks!

Reply:
Actually, the `getDepth()` is defined like this:
https://github.com/dogtagpki/jss/blob/master/base/src/main/java/org/mozilla/jss/ssl/SSLCertificateApprovalCallback.java#L148-L152
so we can always calculate the `depth` as you described from the sorted chain regardless of the original chain.

Reply:
Yes, I read this definition and thought it could be easily calculated but the definition is ambiguous because it is not clear if refer to the provided chain or the real chain